pkg/spanconfig/spanconfigkvsubscriber/testdata/system_span_configs

```diff
@@ -0,0 +1,127 @@
+
+start
+----
+
+update
+upsert [a,c):A
+upsert [d,f):D
+----
+
+updates
+----
+[/Min,/Max)
+[a,c)
+[d,f)
+
+store-reader key=a
+----
+conf=A
+
+store-reader key=e
+----
+conf=D
+
+store-reader key=g
+----
+conf=MISSING
+
+update
+upsert {entire-keyspace}:+X
+----
+
+updates
+----
+[/Min,/Max)
+
+store-reader key=a
+----
+conf=A+X
+
+store-reader key=e
+----
+conf=D+X
+
+# The system span config we set should still apply.
+store-reader key=g
+----
+conf=MISSING+X
+
+# Update span configs that target tenant keyspaces and ensure handlers are
+# invoked correctly.
+update
+upsert {source=1, target=1}:+Y
+upsert {source=1,target=20}:+Z
+upsert {source=10, target=10}:+W
+----
+
+# [/Min,/Tenant/2] corresponds to the system tenant targeting itself.
+updates
+----
+[/Min,/Tenant/2)
+[/Tenant/10,/Tenant/11)
+[/Tenant/20,/Tenant/21)
+
+# Ensure configs are correctly hydrated when we read them.
+store-reader key=a
+----
+conf=A+X+Y
+
+store-reader key=e
+----
+conf=D+X+Y
+
+# Delete the system span config over the entire keyspace and ensure handlers are
+# invoked correctly + configs for various keys are correctly hydrated.
+update
+delete {entire-keyspace}
+----
+
+updates
+----
+[/Min,/Max)
+
+store-reader key=a
+----
+conf=A+Y
```

Comment:
Just looked into the earlier PR that took character tags and int-casted them into protection policies. To me that at least smells weird, I don't know. If all we cared about was unique tags, it's fine to tag them using integers. When printing the configs, instead of this string concat (we're having to use tags prefixed with "+X", which feels odd), we could simply print a list of protection policies in PrintSpanConfig with "+", iterating through each PTS.